### PR TITLE
prompt for username when --username not used

### PR DESCRIPTION
**Pull request by simorgh1**
Signed-off-by: Bahram Maravandi <bahi@hey.com>

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
Please use real name for `Signed-off-by` line

File: cmd/nerdctl/login.go [hunk 1]
```diff
@@ -205,7 +207,17 @@ func ConfigureAuthentification(authConfig *types.AuthConfig, options *loginOptio
 	}
 
 	if options.username == "" {
-		return fmt.Errorf("error: Username is Required")
+		fmt.Printf("Enter Username:")
```
Comment (fahedouch) [inline cmd/nerdctl/login.go:210]: plz add a space suffix and use  `fmt.Print` =>  `fmt.Print("Enter Username: ")`

**Review by fahedouch — CHANGES_REQUESTED** [review]

File: cmd/nerdctl/login.go (hunk 1 above)
Comment (simorgh1) [reply cmd/nerdctl/login.go]: done

**Comment by AkihiroSuda** [comment]
Please squash commits

File: cmd/nerdctl/login.go [hunk 2]
```diff
@@ -205,7 +207,17 @@ func ConfigureAuthentification(authConfig *types.AuthConfig, options *loginOptio
...
-		return fmt.Errorf("error: Username is Required")
+		fmt.Print("Enter Username: ")
+		reader := bufio.NewReader(os.Stdin)
+		usr, err := reader.ReadString('\n')
+		if err != nil {
+			return err
+		}
+		usr = strings.TrimSuffix(usr, "\n")
+		if len(usr) == 0 {
+			return fmt.Errorf("error: Non-null Username Required")
```
Comment (AkihiroSuda) [inline cmd/nerdctl/login.go:218]: "error: " is not needed here

File: cmd/nerdctl/login.go [hunk 3]
```diff
@@ -205,7 +207,17 @@ func ConfigureAuthentification(authConfig *types.AuthConfig, options *loginOptio
...
 
 	if options.username == "" {
-		return fmt.Errorf("error: Username is Required")
+		fmt.Print("Enter Username: ")
+		reader := bufio.NewReader(os.Stdin)
+		usr, err := reader.ReadString('\n')
+		if err != nil {
+			return err
+		}
+		usr = strings.TrimSuffix(usr, "\n")
```
Comment (AkihiroSuda) [inline cmd/nerdctl/login.go:216]: strings.TrimSpace

File: cmd/nerdctl/login.go (hunk 2 above)
Comment (simorgh1) [reply cmd/nerdctl/login.go]: done.

File: cmd/nerdctl/login.go (hunk 3 above)
Comment (simorgh1) [reply cmd/nerdctl/login.go]: Thanks, done.

**Comment by fahedouch** [comment]
we should handle the case where `!terminal.IsTerminal(int(os.Stdin.Fd()))`  , either we don't wait on input or we allocate to tty

**Comment by AkihiroSuda** [comment]
@fahedouch Do you want to open a PR?

**Comment by fahedouch** [comment]
Thanks, carried as https://github.com/containerd/nerdctl/pull/752

**Pull request closed without merging**